src/component/GamePlay/GamePlay.jsx

```diff
@@ -1,56 +1,37 @@
-import React, { useEffect, useState } from "react";
 import "./GamePlay.css";
+import React, { useEffect, useState } from "react";
 import NavBar from "../nav_bar/nav_bar.component";
 import { useLocation } from "react-router-dom";
 
 function GamePlay({ favorites, setFavorites, errorMessage }) {
   const location = useLocation();
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [predatorData, setPredatorData] = useState([]);
-  const [currentAnimal, setCurrentAnimal] = useState(location.state?.rabbitData);
   const [isFavorited, setIsFavorited] = useState(false);
+  const [currentAnimal, setCurrentAnimal] = useState(location.state?.rabbitData);
 
-  console.log("GamePlay mounted with favorites:", favorites);
-  console.log("GamePlay mounted with favorites type:", typeof favorites, Array.isArray(favorites));
+  console.log("Current animal:", currentAnimal);
 
   useEffect(() => {
-    fetchAllFavorites();
-  }, []);
-  
-  useEffect(() => {
-    console.log('Favorites updated to:', favorites);
-  }, [favorites]);
+    const attributes = currentAnimal?.data ? currentAnimal.data[0].attributes : currentAnimal?.attributes;
```

Comment:
This solved the problem of the original currentAnimal(rabbitData) not being able to change the state of the "love" class.